model/clusters_mgmt/v1/external_cluster_service_resource.model

```diff
@@ -0,0 +1,23 @@
+/*
+Copyright (c) 2020 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Manages ExternalClusterService.
+resource ExternalClusterServices {
```

Comment:
ExternalClusterService - single object. Or rename the file.
I think the metamodel will have a problem with giving a resource that is not a collection a plural name

Reply:
It's working. I tested it.
Need to fix the comment though.

Per the filename, there's a problem because the name "external_cluster_services" is the single object here. 
And there's a list of "external_cluster_services", which I called "ExternalClusterServicesCollection".
So I will rename the later filename.

Reply:
Fixed. PreTAL

Reply:
so the endpoint will be `/external_cluster_services` both for root and for cluster?
It is confusing and nonunified. All our other plural endpoints indicate collection.
Maybe we should consider renaming it to something like `external_server_config` (that can be plural `configs` if calling from root)

Reply:
It is confusing, I agree.
It was a bad naming decision on my behalf. I shouldn't have called that single tuple of [hive+awsAcctOperator+gcpProjOperator] something that is plural.

That said, it's already out there. If I want to rename it, I need to do a huge change, which includes configuration changes both on CS basecode, and app-interface. 

I can mitigate that discrepancy by calling the collection of "external-cluster-services" as "external_cluster_services_list", but this also isn't consistent with the other endpoints that we have.

Another option would be to drop the root resource all together, and not expose it at all. You can still see the configuration by querying the DB. And SRE will still be able to see which one is used for their cluster, but won't be able to see all of them using the API.

WDYT?

Reply:
I don't think it's such a big change. You don't need to change everything. Since it's the first time it's exposed in the API - you can determine the way it's called. Maybe `external_cluster_services_config` is not so far from current naming